content/docs/rfcs/32/README.md

```diff
@@ -0,0 +1,457 @@
+Then a secret equation is created in the form of:
+
+```
+y = a_0 + x*a_1,
+```
+
+where `a_0` is the `identity_secret_hash` and `a_1 = hash(a_0, epoch)`.
+Along with the generated proof, 
+the users need to provide a (x, y) share which satisfies the line equation, 
+in order for their proof to be verified. 
+`x` is the hashed signal, while the `y` is the circuit output. 
+With more that the `limit` number of shares anyone can derive `a_0`, the `identity_secret_hash`  key. 
+Hash of a signal will be evaluation point x. 
+So that a member who sends more that `limit` number of signals per epoch reveals the secret key.
+
+Note that shares used in different epochs cannot be used to derive the secret key.
+
+### Nullifiers
+`epoch` is external nullfier.
+
+The internal nullifier is calculated as `nullifier = hash(a_1, rln_identifier)`. 
+Note that `a_1` already contains a secret ingredient `a_0` and epoch ingredient, 
+so each epoch a member can signal to only one nullifier. 
+
+The `rln_identifier` is a random value from a finite field, 
+unique per RLN app, 
+and is used for additional cross-application security - to protect the user secrets being compromised if they use the same credentials accross different RLN apps. 
+If `rln_identifier` is not present, 
+the user uses the same credentials and sends a different message for two different RLN apps using the same epoch,
+then their secret key can be revealed. 
+With adding the `rln_identifier` field we obscure the nullifier, 
+so this kind of attack cannot happen. 
+The only kind of attack that is possible is if we have an entity with a global view of all messages, 
+and they try to brute force different combinations of `x` and `y` shares for different nullifiers.
+
+
+## Identity credentials generation
+
+In order to be able to generate valid proofs, the users need to be part of the identity membership merkle tree. 
+They are part of the identity membership merkle tree if their `identity_commitment` is placed in a leaf in the tree.
+
+The identity credentials of a user are composed of:
+
+- `identity_secret`
+- `identity_secret_hash`
+- `identity_commitment`
+
+### `identity_secret`
+
+The `identity_secret` is generated in the following way:
+
+```
+    identity_nullifier = random_32_byte_buffer
+    identity_trapdoor = random_32_byte_buffer
+    identity_secret = [identity_nullifier, identity_trapdoor]
+```
+
+The same secret should not be used accross different protocols, 
+because revealing the secret at one protocol could break privacy for the user in the other protocols.  
+
+### `identity_secret_hash`
+
+The `identity_secret_hash` is generated by obtaining a Poseidon hash of the `identity_secret` array:
+
+```
+    identity_secret_hash = poseidonHash(identity_secret)
+```
+
+### `identity_commitment`
+
+The `identity_commitment` is generated by obtaining a Poseidon hash of the `identity_secret_hash`: 
+
+```
+identity_commitment = poseidonHash([identity_secret_hash])
+```
+
+
+# Flow
```

Comment:
Perhaps we can make a short list of steps to make the flow more clear? After basic introduction perhaps.

E.g.

1. Registration
2. ...

We can also do a sequence diagram https://rfc.vac.dev/spec/10/#overview-of-protocol-interaction but might be overkill for now, up to you

Reply:
I agree, sequence diagram will be helpful for sure.